control-plane/grpc/proto/v1/volume/volume.proto

```diff
@@ -0,0 +1,270 @@
+  // share protocol
+  optional VolumeShareProtocol share = 3;
+}
+
+// Unpublish a volume from any node where it may be published
+// Unshares the children nexuses from the volume and destroys them.
+message UnpublishVolumeRequest {
+  // uuid of the volume
+  google.protobuf.StringValue uuid = 1;
+  // if the node where the nexus lives is offline then we can force unpublish, forgetting about
+  // the nexus. Note: this option should be used only when we know the node will not become
+  // accessible again and it is safe to do so.
+  bool force = 2;
+}
+
+// Share Volume request
+message ShareVolumeRequest {
+  // uuid of the volume
+  google.protobuf.StringValue uuid = 1;
+  // share protocol
+  VolumeShareProtocol share = 3;
+}
+
+// Unshare Volume request
+message UnshareVolumeRequest {
+  // uuid of the volume
+  google.protobuf.StringValue uuid = 1;
+}
+
+// Set the volume replica count
+message SetVolumeReplicaRequest {
+  // uuid of the volume
+  google.protobuf.StringValue uuid = 1;
+  // replica count
+  uint32 replicas = 2;
+}
+
+// Delete volume
+message DestroyVolumeRequest {
+  // uuid of the volume
+  google.protobuf.StringValue uuid = 1;
+}
+
+// Reply type for a CreateVolume request
+message CreateVolumeReply {
+  oneof reply {
+    Volume volume = 1;
+    common.ReplyError error = 2;
+  }
+}
+
+// Reply type for a GetVolumes request
+message GetVolumesReply {
+  oneof reply {
+    Volumes volumes = 1;
+    common.ReplyError error = 2;
+  }
+}
+
+// Reply type for a DestroyVolume request
+message DestroyVolumeReply {
+  optional common.ReplyError error = 1;
+}
+
+// Reply type for a ShareVolume request
+message ShareVolumeReply {
+  oneof reply {
+    string response = 1;
+    common.ReplyError error = 2;
+  }
+}
+
+// Reply type for a UnshareVolume request
+message UnshareVolumeReply {
+  optional common.ReplyError error = 1;
+}
+
+// Reply type for a PublishVolume request
+message PublishVolumeReply {
+  oneof reply {
+    Volume volume = 1;
+    common.ReplyError error = 2;
+  }
+}
+
+// Reply type for a UnpublishVolume request
+message UnpublishVolumeReply {
+  oneof reply {
+    Volume volume = 1;
+    common.ReplyError error = 2;
+  }
+}
+
+// Reply type for a SetVolumeReplica request
+message SetVolumeReplicaReply {
+  oneof reply {
+    Volume volume = 1;
+    common.ReplyError error = 2;
+  }
+}
+
+service VolumeGrpc {
+  rpc CreateVolume (CreateVolumeRequest) returns (CreateVolumeReply) {}
+  rpc DestroyVolume (DestroyVolumeRequest) returns (DestroyVolumeReply) {}
+  rpc GetVolumes (GetVolumesRequest) returns (GetVolumesReply) {}
+  rpc PublishVolume (PublishVolumeRequest) returns (PublishVolumeReply) {}
+  rpc UnpublishVolume (UnpublishVolumeRequest) returns (UnpublishVolumeReply) {}
+  rpc ShareVolume (ShareVolumeRequest) returns (ShareVolumeReply) {}
+  rpc UnshareVolume (UnshareVolumeRequest) returns (UnshareVolumeReply) {}
+  rpc SetVolumeReplica (SetVolumeReplicaRequest) returns (SetVolumeReplicaReply) {}
+}
```

Comment:
Should we start adding readiness probes? Though I wonder if that should be a separate service.

Reply:
We can probab;y once each of these features are ready.